addons/hr_holidays/static/src/views/calendar/common/calendar_common_popover.js

```diff
@@ -1,25 +1,38 @@
-/** @odoo-module */
-
 import { CalendarCommonPopover } from "@web/views/calendar/calendar_common/calendar_common_popover";
-
+import { onWillStart } from "@odoo/owl";
+import { user } from "@web/core/user";
 import { useService } from "@web/core/utils/hooks";
 
 export class TimeOffCalendarCommonPopover extends CalendarCommonPopover {
+    static subTemplates = {
+        ...CalendarCommonPopover.subTemplates,
+        footer: "hr_holidays.TimeOffCalendarCommonPopover.footer",
+    };
+
     setup() {
         super.setup();
-
-        this.dialog = useService('dialog');
-        this.action = useService('action');
+        this.orm = useService("orm");
+        this.viewType = "calendar";
+        onWillStart(async () => {
+            this.record = this.props.record.rawRecord
+            this.state = this.record.state;
+            this.isManager = await user.hasGroup("hr_holidays.group_hr_holidays_user") || this.record.leave_manager_id?.[0] == user.userId;
+        });
     }
 
     get isEventDeletable() {
-        const record = this.props.record.rawRecord;
-        const state = record.state;
-        return record.can_cancel || state && !['validate', 'refuse', 'cancel'].includes(state);
+        return this.props.record.rawRecord.can_cancel || this.state && !['validate', 'refuse', 'cancel'].includes(this.state);
     }
 
     get isEventEditable() {
-        const state = this.props.record.rawRecord.state;
-        return state !== undefined;
+        return this.state !== undefined;
+    }
+
+    async onClickButton(ev) {
+        const args = (ev.target.name === "action_approve") ? [this.record.id, false] : [this.record.id];
+        await this.orm.call("hr.leave", ev.target.name, args);
+        await this.props.model.fetchRecords(this.props.model.data);
```

Comment:
Why calling this function here?
It's called when executing `model.load()` and also its results are not used.
Looks like it's useless to me.

Reply:
Hello @mcm-odoo @Bertrand2 

Ohh you are right the load method will do the work here
there is no need to `fetchRecords` my bad :pray: 

I've applied the changes :innocent: